cli/install/template.go

```diff
@@ -154,6 +154,8 @@ spec:
         {{.CreatedByAnnotation}}: {{.CliVersion}}
     spec:
       serviceAccount: linkerd-controller
+      securityContext:
+        runAsUser: {{.ControllerUID}}
```

Comment:
I don't think we can set `securityContext` at the pod level, and instead we'll need to see it at the container level. Setting it at the pod level applies to all containers in the pod, including init containers, and as far as I can tell the `linkerd-init` init container will exit silently with an error if it's not run as root.

To repro with this branch, I ran:

```
bin/fast-build
bin/linkerd install | kubectl apply -f -
kubectl -n linkerd logs deploy/linkerd-controller linkerd-init
2018/12/07 23:38:38 Tracing this script execution as [1544225918]
2018/12/07 23:38:38 State of iptables rules before run:
2018/12/07 23:38:38 > iptables -t nat -vnL
2018/12/07 23:38:38 < iptables v1.4.21: can't initialize iptables table `nat': Permission denied (you must be root)
Perhaps iptables or your kernel needs to be upgraded.

2018/12/07 23:38:38 Aborting firewall configuration
Error: exit status 3
Usage:
  proxy-init [flags]

Flags:
  -h, --help                                help for proxy-init
      --inbound-ports-to-ignore intSlice    Inbound ports to ignore and not redirect to proxy. This has higher precedence than any other parameters.
  -p, --incoming-proxy-port int             Port to redirect incoming traffic (default -1)
      --outbound-ports-to-ignore intSlice   Outbound ports to ignore and not redirect to proxy. This has higher precedence than any other parameters.
  -o, --outgoing-proxy-port int             Port to redirect outgoing traffic (default -1)
  -r, --ports-to-redirect intSlice          Port to redirect to proxy, if no port is specified then ALL ports are redirected
  -u, --proxy-uid int                       User ID that the proxy is running under. Any traffic coming from this user will be ignored to avoid infinite redirection loops. (default -1)
      --simulate                            Don't execute any command, just print what would be executed
```

Moving this `securityContext` attribute to each of the containers fixes the issue, but that also means that the `linkerd-init` container will continue as run as root, in which case this branch won't be a complete fix for #1930. But I'm not sure it's going to be possible to run the init container under a different UID, since it mucks with iptables.

Reply:
Note that the `linkerd-init` container already sets its security context as follows:

```
        securityContext:
          capabilities:
            add:
            - NET_ADMIN
          privileged: false
```

Reply:
Good catch! I have updated the initContainer to `runAsUser: 0`, `runAsGroup: 0` and `runAsNonRoot: false`. This looks like it fixes the startup problem you saw. I will amend the issue: #1930 to recognize that the `initContainer`s will need root no matter what, until #1887 is resolved.